libs/gen/EnumsSourceGenerator.cs

```diff
@@ -0,0 +1,151 @@
+            classBuilder.AppendLine();
+            classBuilder.AppendLine("/// <summary>");
+            classBuilder.AppendLine($"/// Utility methods for enums.");
+            classBuilder.AppendLine("/// </summary>");
+            classBuilder.AppendLine($"public static partial class {GeneratedClassName}");
+            classBuilder.AppendLine("{");
+            classBuilder.AppendLine(GenerateTryParseEnumFromDescriptionMethod(enumName, values));
+            classBuilder.AppendLine();
+            classBuilder.AppendLine(GenerateGetEnumDescriptionsMethod(enumName, values));
+            classBuilder.AppendLine("}");
+            var classSource = classBuilder.ToString();
+            context.AddSource($"{GeneratedClassName}.{enumName}.g.cs", classSource);
+        }
+    }
+
+    private static string GenerateTryParseEnumFromDescriptionMethod(string enumName, List<(string Name, string? Value, AttributeSyntax Description)> values)
+    {
+        var method = new StringBuilder();
+        method.AppendLine($"    /// <summary>");
+        method.AppendLine($"    /// Tries to parse the enum value from the description.");
+        method.AppendLine($"    /// </summary>");
+        method.AppendLine($"    /// <param name=\"description\">Enum description.</param>");
+        method.AppendLine($"    /// <param name=\"result\">Enum value.</param>");
+        method.AppendLine($"    /// <returns>True if successful.</returns>");
+        method.AppendLine($"    public static bool TryParse{enumName}FromDescription(string description, out {enumName} result)");
+        method.AppendLine("    {");
+        method.AppendLine("        result = default;");
+        method.AppendLine("        switch (description)");
+        method.AppendLine("        {");
+        foreach (var (name, _, description) in values)
+        {
+            bool hasDescription = false;
+            var descriptionValue = description?.ArgumentList?.Arguments.FirstOrDefault()?.ToString();
+            if (descriptionValue is not null)
+            {
+                hasDescription = true;
+                method.AppendLine($"            case {descriptionValue}:");
+            }
+
+            if (!hasDescription) continue;
+            method.AppendLine($"                result = {enumName}.{name};");
+            method.AppendLine("                return true;");
+        }
+        method.AppendLine("        }");
+        method.AppendLine();
+        method.AppendLine("        return false;");
+        method.AppendLine("    }");
+        method.AppendLine();
+
+        return method.ToString();
+    }
+    private static string GenerateGetEnumDescriptionsMethod(string enumName, List<(string Name, string? Value, AttributeSyntax Description)> values)
+    {
+        var method = new StringBuilder();
+        method.AppendLine($"    /// <summary>");
+        method.AppendLine($"    /// Gets the descriptions of the set flags. Assumes the enum is a flags enum.");
```

Comment:
I noticed that only 3 enums use this utility class and all 3 have the `Flags` attribute